### PR TITLE
Delete function

### PR DESCRIPTION
**Pull request by nkosi-tauro**
Added Delete function

- Has dialog for user to confirm if they want to go ahead and delete

- DeleteRow method added to DatabaseHelper

Updated gif showing delete in action
Android-Emulator-Pixel-2-API-29

### CODE REVIEW

File: app/src/main/java/com/app/courseplan/CourseAdapter.java [hunk 1]
```diff
@@ -40,10 +43,20 @@ public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
...
+        holder.mainLayout.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+//                this to to get the course Id and name and send them to the Course Details Class for the delete dialog
+                Intent intent = new Intent(context, CourseDetails.class);
+                intent.putExtra("id", String.valueOf(course.getId()));
+                intent.putExtra("title", String.valueOf(course.getCourseName()));
+                context.startActivity(intent);
+            }
+        });
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/CourseAdapter.java]: I already implemented this

File: app/src/main/java/com/app/courseplan/ui/CourseDetails.java [hunk 2]
```diff
@@ -141,6 +155,31 @@ public void onDateSet(DatePicker datePicker, int year, int month, int day) {
 
         }
 
+
+    }
+
+    // Confirm Dialog for delete button
+    void confirmDialog() {
+        id = getIntent().getStringExtra("id");
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/ui/CourseDetails.java]: You can get the course object with all data fields from getParcelableObject("seelected_course") instead
Check MainActivity line 175 https://github.com/Acrylami/CodingCrowsTeam3/blob/master/app/src/main/java/com/app/courseplan/ui/MainActivity.java#L175

**Review by redatawfik — COMMENTED** [review]
Can you add a screenshot for delete button?

File: app/src/main/java/com/app/courseplan/CourseAdapter.java (hunk 1 above)
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/CourseAdapter.java]: Remove

File: app/src/main/java/com/app/courseplan/CourseAdapter.java [hunk 3]
```diff
@@ -61,12 +74,14 @@ public int getItemCount() {
         private TextView courseId;
         private TextView courseTitle;
         private TextView courseUrl;
+        LinearLayout mainLayout;
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/CourseAdapter.java]: remove

File: app/src/main/java/com/app/courseplan/CourseAdapter.java [hunk 4]
```diff
@@ -61,12 +74,14 @@ public int getItemCount() {
...
 
         public MyViewHolder(@NonNull View itemView) {
             super(itemView);
             courseTitle = itemView.findViewById(R.id.course_title_txt);
             courseId = itemView.findViewById(R.id.course_id_txt);
             courseUrl = itemView.findViewById(R.id.course_url_txt);
+            mainLayout = itemView.findViewById(R.id.mainLayout);
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/CourseAdapter.java]: remove

File: app/src/main/java/com/app/courseplan/ui/CourseDetails.java [hunk 5]
```diff
@@ -20,9 +22,10 @@
 
 public class CourseDetails extends AppCompatActivity {
 
-    Button cancelButton;
-    Button saveButton;
+    Button cancelButton, saveButton, deleteButton;
     EditText courseTitle, courseURL, courseDescription, startDate, endDate;
+    //Course ID/ Title for the delete dialog
+    String id, title;
     //    Date Picker
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/ui/CourseDetails.java]: private?

File: app/src/main/java/com/app/courseplan/ui/CourseDetails.java [hunk 6]
```diff
@@ -141,6 +155,31 @@ public void onDateSet(DatePicker datePicker, int year, int month, int day) {
...
+    void confirmDialog() {
+        id = getIntent().getStringExtra("id");
+        title = getIntent().getStringExtra("title");
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setTitle("Delete " + title + " ?");
+        builder.setMessage("Are you sure you want to delete " + title + " ?");
+        builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int i) {
+                DatabaseHelper myDB = new DatabaseHelper(CourseDetails.this);
```
Comment (redatawfik) [inline app/src/main/java/com/app/courseplan/ui/CourseDetails.java:173]: I see you're using DatabaseHelper in many places, you can move it to class members

**Comment by nkosi-tauro** [comment]
> Can you add a screenshot for delete button?

@redatawfik  I put the screen record on discord as github doesn't want to play it

Noted with regards to feedback, I will Implement the changes.

File: app/src/main/java/com/app/courseplan/ui/CourseDetails.java (hunk 6 above)
Comment (nkosi-tauro) [reply app/src/main/java/com/app/courseplan/ui/CourseDetails.java:173]: Noted, looking into this.

**Review by Acrylami — APPROVED** [review]
Looks good to me

**Comment by redatawfik** [comment]
Looks good to me, Please update your local repo to solve the merge conflict

**Review by redatawfik — APPROVED** [review]

**Pull request merged**